### DISCUSSION TITLE
Nice mesh with periodic boundaries

### DISCUSSION THREAD

**Discussion by IdaRlt** [neperfepx/neper#757] [Q&A] [answered]
Hello all,
I understand that the regularization option is not implemented for periodic tessellations. Is there any workaround to get a nice mesh anyway ? 
Thanks.

**Comment by rquey** [comment, marked as answer]
1. Use a morphology other than Voronoi (with more spherical cells)
2. Use `-pl 100` (or more) in `-M`

**Comment by IdaRlt** [comment]
@rquey, thank you very much for your quick answer.
I had been playing with the -pl option but not sphericity. I don't know yet the actual morphology of my material, and the reasonable range for this parameter, hence I had left the default option.
Increasing sphericity indeed reduces a lot the occurrence of small edges (although, I guess that it is impossible to avoid them completely).

**Reply by mkasemer** [reply]
You could increasingly avoid by:

1. Both increasing the mean sphericity (as Romain suggested, or more accurately for Neper's input of `1-s`, you should be focusing on an increasingly smaller value toward 0), but also reducing the variability of sphericity (reducing the standard deviation).
2. Forcing the grains to be increasingly equally-sized.

The extreme of this case would be approaching the "centroidal" tessellation. You will likely find fewer small edges in these types of microstructures (but on the other hand they become more cartoonish / less based on physical observations).

**Comment by IdaRlt** [comment]
Yes, indeed, and this means representing a virtual  microstructure that might be significantly different from real life.  Investigating the effect of the grain size distribution, in particular, is interesting in order to understand fatigue issues. 
I was wondering : why isn't regularization implemented for periodic boundaries ?  Is it just that nobody ever had the time (I guess it could be cumbersome - several special cases to deal with) or are there some more fundamental difficulties ?

**Reply by rquey** [reply]
If you take a grain growth structure (`gg`), it is both representative of real life (more than a Voronoi tessellation, which is the first case I dealt with (for regularization)) and has much less geometrical details than VT. See Figure 20a of https://neper.info/papers.html#id2.  So, you don't necessarily have to go the the extreme case of centroidal VT.

Regularization of periodic structures has simply never made it to the top of the TODO list, but it will soon :).
